### PR TITLE
Helm CRUD release

### PR DESCRIPTION
**Pull request by markfuller**
Add a CRUD-able helm resource, called Release, in place of the helm action.

### CODE REVIEW

File: cmd/goplugin-helm/helm/start.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		s := Server(c)
+		grpc.Serve(c, s)
+	})
+}
+
+// Server returns the built server to be served
+func Server(c px.Context) *service.Server {
+	sb := service.NewServiceBuilder(c, "Helm")
+
+	evs := sb.RegisterTypes("Helm", resource.Release{})
```
Comment (thallgren) [inline cmd/goplugin-helm/helm/start.go]: Shouldn't "Name" be annotated as an `immutableAttribute` here? Something like:
```Go
evs := sb.RegisterTypes("Helm", 
  sb.BuildResource(&resource.Release{}, func(rb service.ResourceTypeBuilder) {
    rb.ImmutableAttributes(`name`)
  }))
```

File: cmd/goplugin-helm/resource/release.go [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	}
+	//an empty response means no releases
+	if len(s) == 0 {
+		return nil, serviceapi.NotFound("release", externalID)
+	}
+	var releaseList ReleaseList
+	b := []byte(s)
+	err = json.Unmarshal(b, &releaseList)
+	if err != nil {
+		panic(fmt.Errorf("unable to unmarshal, err is %v and output is %s", err, b))
```
Comment (thallgren) [inline cmd/goplugin-helm/resource/release.go]: Why use panic here but `return nil, err` on line 52?

File: cmd/goplugin-helm/resource/release.go [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	}
+	_, err := runCmd("", "helm", args...)
+	log.Debug("completed helm upgrade", "err", err)
+	if err != nil {
+		return nil, err
+	}
+	return h.Read(extID(*desiredState))
+}
+
+func runCmd(dir string, name string, arg ...string) ([]byte, error) {
```
Comment (thallgren) [inline cmd/goplugin-helm/resource/release.go]: Nit, but `arg` should probably be `args` here too.

File: cmd/goplugin-helm/resource/release.go (hunk 2 above)
Comment (markfuller) [reply cmd/goplugin-helm/resource/release.go]: agreed, left in by mistake.

File: cmd/goplugin-helm/resource/release.go (hunk 3 above)
Comment (markfuller) [reply cmd/goplugin-helm/resource/release.go]: yep.

File: cmd/goplugin-helm/helm/start.go (hunk 1 above)
Comment (markfuller) [reply cmd/goplugin-helm/helm/start.go]: 👍

**Pull request merged**